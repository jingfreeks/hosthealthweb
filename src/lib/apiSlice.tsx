import { setCredentials, setLogout } from "@/slice/auth";
import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";

const baseQuery = fetchBaseQuery({
  baseUrl: process.env.DEV_BASE_URL,
  credentials: "include",
  prepareHeaders: (headers, { getState }: any) => {
    const token = getState().auth.token;
    if (token) {
      headers.set("authorization", `Bearer ${token}`);
    }
    return headers;
  },
});

const baseQueryWithAuth = async (
  args: string,
  api: any,
  extraOptions: string
) => {
  let result: any = await baseQuery(args, api, extraOptions);
  if (result?.error?.originalStatus === 403) {
    const refereshResult = await baseQuery("/refresh", api, extraOptions);

    if (refereshResult?.data) {
      const user = api.getState().auth.user;
      //store the new token
      api.dispatch(setCredentials({ ...refereshResult.data, user }));
      //retry the original query with new access token
      result = await baseQuery(args, api, extraOptions);
    } else {
      api.dispatch(setLogout());
    }
  }
  return result;
};

export const apiSlice = createApi<any, any>({
  baseQuery: baseQueryWithAuth,
  reducerPath: "api",
  endpoints: (builder) => ({}),
  tagTypes: ["City", "Jobs", "MyJobs", "JobDetails", "Profile"] as any,
});
