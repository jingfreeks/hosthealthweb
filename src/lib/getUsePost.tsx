import React from "react";

export default async function getUserPosts(userId: string) {
  const response = await fetch(
    `https://jsonplaceholder.typicode.com/posts?userId=${userId}`,
    { next: { revalidate: 60 } }
  );

  if (!response.ok) undefined;
  return response.json();
}
