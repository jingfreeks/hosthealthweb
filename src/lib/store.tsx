import { configureStore, combineReducers } from "@reduxjs/toolkit";
import authReducer from "@/slice/auth";
import {apiSlice} from './apiSlice';

const rootReducer = combineReducers({
  [apiSlice.reducerPath]: apiSlice.reducer,
  auth: authReducer,
});
export const setupStore= (preloadedState?: Partial<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    middleware: getDefaultMiddleware =>
        getDefaultMiddleware({ serializableCheck: false}).concat(apiSlice.middleware),
      devTools: true,
    preloadedState
  });
};

// Infer the type of makeStore
export type AppStore = ReturnType<typeof setupStore>;
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = AppStore["dispatch"];
