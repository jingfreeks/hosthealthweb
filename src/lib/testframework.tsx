/* eslint-disable @typescript-eslint/no-explicit-any */
import React,{PropsWithChildren} from 'react';
import {render} from '@testing-library/react';
import type {RenderOptions} from '@testing-library/react';
import {Provider} from 'react-redux';
import type {AppStore, RootState} from './store';
import {setupStore} from './store'
// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
