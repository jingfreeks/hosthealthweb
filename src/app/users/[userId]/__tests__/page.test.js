import { render } from "@testing-library/react";
import {renderWithProviders} from '../../../../lib/testframework';
import Page from "../page";

describe("User Id Screen Component Page", () => {
  it("renders User Id Screen Component Page to match snaptshots", async () => {
    const { container } = renderWithProviders(
      await Page({
        params: {
          userId: "test user",
        },
      })
    );
    expect(container).toMatchSnapshot();
  });
});
