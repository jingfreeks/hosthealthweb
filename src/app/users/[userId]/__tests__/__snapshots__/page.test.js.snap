// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Id Screen Component Page renders User Id Screen Component Page to match snaptshots 1`] = `<div />`;
